src/app/model/King.ts

```diff
@@ -0,0 +1,27 @@
+import { PiecePositions } from './PiecesPositions';
+import { Piece } from './Piece';
+import { columns, Row, Side, Square } from './Types';
+import { PiecePossibleMoves } from './PiecePossibleMoves';
+
+export class King extends Piece implements PiecePossibleMoves {
+  constructor(side: Side) {
+    super(side);
+  }
+
+  possibleMoves(position: Square, board: PiecePositions): Square[] {
+    const currentColumnNumber = columns.indexOf(position.column);
+    const currentRowNumber = position.row;
+    return [
+      { column: columns[currentColumnNumber - 1], row: (currentRowNumber - 1) as Row },
+      { column: columns[currentColumnNumber - 1], row: currentRowNumber as Row },
+      { column: columns[currentColumnNumber - 1], row: (currentRowNumber + 1) as Row },
+      { column: columns[currentColumnNumber], row: (currentRowNumber - 1) as Row },
+      { column: columns[currentColumnNumber], row: (currentRowNumber + 1) as Row },
+      { column: columns[currentColumnNumber + 1], row: (currentRowNumber - 1) as Row },
+      { column: columns[currentColumnNumber + 1], row: currentRowNumber as Row },
+      { column: columns[currentColumnNumber + 1], row: (currentRowNumber + 1) as Row },
+    ]
+      .filter(King.isWithinChessboardBorders)
+      .filter((square) => this.checkIfOponent(square, board));
```

Comment:
Coś tutaj jest dziwnego jak czytam ten kod, to mi wychodzi: To ruchy wymienione, tylko te które są w granicach planszy i te na których stoi przeciwnik. Jak to działa  :P ?

Reply:
Co z ruchami gdzie nie stoi nic?

Reply:
Z tego co mi sie wydaje, chyba jest blad w checkIfOponent i dlatego to dziala.

Reply:
ciekawa uwaga

Reply:
elegancko jest

Reply:
To jak dla mnie zle dziala to checkIfOponent, logicznie by bylo jakby zwracala true TYLKO jesli stoi tam bierka przeciwnika, a wprzeciwnym wypadku false. Nie sadzicie? Teraz ten kod jest nie zrozumialy.

Reply:
ale w środku ta metoda korzysta z onPositionPiece, które zwraca Piece albo undeifned, więc musi być ten '?'

Reply:
To może zmienić nazwę tej metody checkIfOponent na checkIfNotSameColorPiece? ;)